baseapp/baseapp.go

```diff
@@ -585,20 +585,43 @@ func getState(app *BaseApp, mode runTxMode) *state {
 
 func (app *BaseApp) initializeContext(ctx sdk.Context, mode runTxMode) sdk.Context {
 	if mode == runTxModeSimulate {
-		ctx = ctx.WithMultiStore(getState(app, runTxModeSimulate).CacheMultiStore())
+		ctx = ctx.WithMultiStore(app.getState(runTxModeSimulate).CacheMultiStore())
 	}
 	return ctx
 }
 
+// cacheTxContext returns a new context based off of the provided context with a
+// cache wrapped multi-store and the store itself to allow the caller to write
+// changes from the cached multi-store. If cached multi-store will also have
+// tracing if enabled.
+func (app *BaseApp) cacheTxContext(
```

Comment:
Should we just return the `msCache.write` function?

Reply:
(great DRY here though)

Reply:
I took a look at the public API of `msCache` (`sdk.CacheMultiStore`) and I don't see anything that could be of any direct harm if called. What do you see as the benefit here?

Reply:
It's just a bit cleaner since we don't expect the caller to use any other functions. Not a big deal though.

Reply:
I think we should leave it for now unless it can seem to cause bugs easily.